nncf/data/dataset.py

```diff
@@ -0,0 +1,118 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Iterable
+from typing import Any
+from typing import Callable
+from typing import Optional
+from typing import List
+
+
+DataItem = Any
```

Comment:
Use `TypeVar` instead of `Any`, because `Any` in different places within the same signature will match to objects of different type, while I think you need `DataItem` to represent one and the same type within one signature.

Reply:
Done